src/Abstractions/TaskName.cs

```diff
@@ -8,127 +8,162 @@ namespace Microsoft.DurableTask;
+  /// Implicitly converts a <see cref="TaskName"/> into a <see cref="string"/> of the <see cref="Name"/> property value.
+  /// </summary>
+  /// <param name="value">The <see cref="TaskName"/> to be converted into a string.</param>
+  public static implicit operator string(TaskName value) => value.Name;
 
-    /// <summary>
-    /// Compares two <see cref="TaskName"/> objects for equality.
-    /// </summary>
-    /// <param name="a">The first <see cref="TaskName"/> to compare.</param>
-    /// <param name="b">The second <see cref="TaskName"/> to compare.</param>
-    /// <returns><c>true</c> if the two <see cref="TaskName"/> objects are equal; otherwise <c>false</c>.</returns>
-    public static bool operator ==(TaskName a, TaskName b)
-    {
-        return a.Equals(b);
-    }
+  /// <summary>
+  /// Implicitly converts a <see cref="string"/> into a <see cref="TaskName"/> value.
+  /// </summary>
+  /// <param name="value">The string to convert into a <see cref="TaskName"/>.</param>
+  public static implicit operator TaskName(string value) => string.IsNullOrEmpty(value) ? default : new(value);
 
-    /// <summary>
-    /// Compares two <see cref="TaskName"/> objects for inequality.
-    /// </summary>
-    /// <param name="a">The first <see cref="TaskName"/> to compare.</param>
-    /// <param name="b">The second <see cref="TaskName"/> to compare.</param>
-    /// <returns><c>true</c> if the two <see cref="TaskName"/> objects are not equal; otherwise <c>false</c>.</returns>
-    public static bool operator !=(TaskName a, TaskName b)
-    {
-        return !a.Equals(b);
-    }
+  /// <summary>
+  /// Compares two <see cref="TaskName"/> objects for equality.
+  /// </summary>
+  /// <param name="a">The first <see cref="TaskName"/> to compare.</param>
+  /// <param name="b">The second <see cref="TaskName"/> to compare.</param>
+  /// <returns><c>true</c> if the two <see cref="TaskName"/> objects are equal; otherwise <c>false</c>.</returns>
+  public static bool operator ==(TaskName a, TaskName b)
+  {
+    return a.Equals(b);
+  }
+
+  /// <summary>
+  /// Compares two <see cref="TaskName"/> objects for inequality.
+  /// </summary>
+  /// <param name="a">The first <see cref="TaskName"/> to compare.</param>
+  /// <param name="b">The second <see cref="TaskName"/> to compare.</param>
+  /// <returns><c>true</c> if the two <see cref="TaskName"/> objects are not equal; otherwise <c>false</c>.</returns>
+  public static bool operator !=(TaskName a, TaskName b)
+  {
+    return !a.Equals(b);
+  }
+
+  /// <summary>
+  /// Gets a value indicating whether to <see cref="TaskName"/> objects
+  /// are equal using value semantics.
+  /// </summary>
+  /// <param name="other">The other object to compare to.</param>
+  /// <returns><c>true</c> if the two objects are equal using value semantics; otherwise <c>false</c>.</returns>
+  public bool Equals(TaskName other)
+  {
+    return string.Equals(this.Name, other.Name, StringComparison.OrdinalIgnoreCase);
```

Comment:
We will need to update this to include `Version` in the comparison.

Reply:
Updated as your suggestion.